controllers/usersignup/usersignup_controller.go

```diff
@@ -282,39 +291,41 @@ func (r *Reconciler) checkIfMurAlreadyExists(reqLogger logr.Logger, config toolc
 			if err := r.setStateLabel(reqLogger, userSignup, toolchainv1alpha1.UserSignupStateLabelValueDeactivated); err != nil {
 				return true, err
 			}
+
 			// We set the inProgressStatusUpdater parameter here to setStatusDeactivationInProgress, as a temporary status before
 			// the main reconcile function completes the deactivation process
-			reqLogger.Info("deleting MasterUserRecord since user has been deactivated")
-			return true, r.deleteMasterUserRecord(mur, userSignup, reqLogger, r.setStatusDeactivationInProgress, r.setStatusFailedToDeleteMUR)
+			reqLogger.Info("Deleting MasterUserRecord since user has been deactivated")
+			return true, r.deleteMasterUserRecord(mur, userSignup, reqLogger, r.setStatusDeactivationInProgress)
 		}
 
 		// if the UserSignup doesn't have the state=approved label set, then update it
 		if err := r.setStateLabel(reqLogger, userSignup, toolchainv1alpha1.UserSignupStateLabelValueApproved); err != nil {
 			return true, err
 		}
 
-		// look-up the default NSTemplateTier to get the NS templates
-		nstemplateTier, err := getNsTemplateTier(r.Client, config.Tiers().DefaultTier(), userSignup.Namespace)
+		// look-up the default NSTemplateTier to set it on the MUR if not set
+		defaultTier, err := getNsTemplateTier(r.Client, config.Tiers().DefaultTier(), userSignup.Namespace)
 		if err != nil {
 			return true, r.wrapErrorWithStatusUpdate(reqLogger, userSignup, r.setStatusNoTemplateTierAvailable, err, "")
 		}
 
 		// check if anything in the MUR should be migrated/fixed
-		if changed, err := migrateOrFixMurIfNecessary(mur, nstemplateTier, userSignup); err != nil {
-			return true, r.wrapErrorWithStatusUpdate(reqLogger, userSignup, r.setStatusInvalidMURState, err, "unable to migrate or fix existing MasterUserRecord")
-
-		} else if changed {
-			reqLogger.Info("updating MasterUserRecord after it was migrated")
+		if changed := migrateOrFixMurIfNecessary(mur, defaultTier, userSignup); changed {
+			reqLogger.Info("Updating MasterUserRecord after it was migrated")
 			if err := r.Client.Update(context.TODO(), mur); err != nil {
-				return true, r.wrapErrorWithStatusUpdate(reqLogger, userSignup, r.setStatusInvalidMURState, err, "unable to migrate or fix existing MasterUserRecord")
+				return true, r.wrapErrorWithStatusUpdate(reqLogger, userSignup, r.setStatusInvalidMURState, err, "unable update MasterUserRecord to complete migration")
 			}
 			return true, nil
 		}
+		reqLogger.Info("MasterUserRecord exists", "name", mur.Name)
+
+		if shouldManageSpace(userSignup) {
+			if err = r.ensureSpace(reqLogger, userSignup, mur); err != nil {
+				return true, err
+			}
```

Comment:
Thanks for moving it here, but the `ensureSpace` method should return also information if the `Space` was created or not. If it was, then we should return here and wait for the next reconcile:
```go
if created, err = r.ensureSpace(reqLogger, userSignup, mur); created || err != nil {
    return true, err
}
```

Reply:
Sorry if this is a stupid question but what would be the benefit/reason for doing that rather than just continuing through the Reconcile? 😅 One thing to add is I've updated this area of the code so the `r.ensureSpaceBinding()` is called right after `r.ensureSpace` so not sure if that changes anything.

Reply:
We should do only one watched change in one reconcile. We do watch Space & SpaceBinding, so when we create the Space, then we should return and wait for the next reconcile. Then when we create SpaceBinding we should again return and wait for the next reconcile. 
When everything is created - in the last reconcile - then we should set the complete status to true.

Reply:
Done in https://github.com/codeready-toolchain/host-operator/pull/595/commits/da6c8fcd54fc44d677ae5b3125d0bafb24d12a11 sorry for the delay